src/main/scala/com/codacy/docker/api/metrics/DockerMetrics.scala

```diff
@@ -0,0 +1,73 @@
+package com.codacy.docker.api.metrics
+
+import codacy.docker.api.metrics.{FileMetrics, MetricsTool}
+import codacy.docker.api.{MetricsConfiguration, Source}
+import com.codacy.api.dtos.Language
+import com.codacy.docker.api.metrics.DockerMetricsEnvironment._
+import com.codacy.docker.api.utils.Delayed
+
+import scala.concurrent.Future
+import scala.concurrent.duration._
+import scala.util.{Failure, Success, Try}
+abstract class DockerMetrics(metricsTool: MetricsTool) extends Delayed {
```

Comment:
Pass env as parameter.

Use env instead of properties.

Reply:
Pass also the printer

Reply:
✅